tests/s3/versioning/test_versioning_multipart_part_2.py

```diff
@@ -0,0 +1,228 @@
+
+    @pytest.mark.parallel
+    @pytest.mark.s3_versioning
+    @pytest.mark.s3_ops
+    @pytest.mark.tags('TEST-41289')
+    def test_upload_multiple_versions_to_multipart_uploaded_object_in_versioned_bucket_41289(self):
+        """Test Upload multiple versions to a multipart uploaded object in a versioned bucket."""
+        self.log.info("STARTED: Test Upload multiple versions to a multipart uploaded object in "
+                      "a versioned bucket.")
+        versions = {}
+        self.log.info("Step 1: Create bucket.")
+        assert_utils.assert_in(self.bucket_name, self.bkt_list,
+                               f"Bucket '{self.bucket_name}' does not exists")
+        self.log.info("Step 2: PUT Bucket versioning with status as Enabled.")
+        self.s3ver_test_obj.put_bucket_versioning(self.bucket_name, status="Enabled")
+        self.log.info("Step 3: Upload a multipart object - mpobject1: mpversionid1")
+        s3ver_cmn_lib.upload_version(self.s3mp_test_obj, self.bucket_name, self.object_name,
+                                     self.test_file_path, versions_dict=versions,
+                                     is_multipart=True, total_parts=10, file_size=100)
+        self.log.info("Step 4: Upload 4 more versions to mpobject1 and then multipart upload.")
+        for i in range(1, 4):
+            system_utils.create_file(self.test_file_path, count=10 * i)
+            s3ver_cmn_lib.upload_version(self.s3test_obj, self.bucket_name, self.object_name,
+                                         self.test_file_path, versions_dict=versions)
+        s3ver_cmn_lib.upload_version(self.s3mp_test_obj, self.bucket_name, self.object_name,
+                                     self.test_file_path, versions_dict=versions,
+                                     is_multipart=True, total_parts=10, file_size=200)
+        self.log.info("Step 5: PUT Bucket versioning with status as Suspended.")
+        self.s3ver_test_obj.put_bucket_versioning(self.bucket_name, status="Suspended")
+        self.log.info("Step 6: Perform PUT Object to mpobject1 (version with 'versionId=null').")
+        s3ver_cmn_lib.upload_version(self.s3test_obj, self.bucket_name, self.object_name,
+                                     self.test_file_path, versions_dict=versions,
+                                     chk_null_version=True)
+        self.log.info("Step 7: Perform List Object Versions.")
+        s3ver_cmn_lib.check_list_object_versions(self.s3ver_test_obj, self.bucket_name, versions)
+        self.log.info("Step 8: Perform List Objects.")
+        resp = self.s3test_obj.object_list(self.bucket_name)
+        assert_utils.assert_true(resp[0], resp[1])
+        assert_utils.assert_in(self.object_name, resp[1],
+                               f"Failed to list versioned object {self.object_name}")
+        self.log.info("Step 9: Perform GET/HEAD Object.")
+        s3ver_cmn_lib.check_get_head_object_version(self.s3test_obj, self.s3ver_test_obj,
```

Comment:
You would need to pass etag of the latest version here. Etag and latest version id can be fetched from the versions dict